src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -202,14 +202,26 @@ public override long Position
         public override void Flush()
         {
             EnsureNotDisposed();
+            FlushBuffers();
         }
 
         public override Task FlushAsync(CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
-            return cancellationToken.IsCancellationRequested ?
-                Task.FromCanceled(cancellationToken) :
-                Task.CompletedTask;
+            return cancellationToken.IsCancellationRequested ? Task.FromCanceled(cancellationToken) : FlushAsyncCore(cancellationToken);
```

Comment:
It is common to call Flush/FlushAsync when mode != Compress?  If so, that can be special cased here rather than kicking off the async operation that'll end up just bailing when it sees the mode != Compress, e.g.
```C#
return
    cancellationToken.IsCancellationRequested ? Task.FromCanceled(cancellationToken) :
    _mode != CompressionMode.Compress ? Task.CompletedTask :
    FlushAsyncCore(cancellationToken);
```
If it's not common, then it's not worth special-casing.